vision/fiducial_detection_balsaq.cc

```diff
@@ -25,11 +25,21 @@ void FidicualDetectionBq::loop() {
   CameraImageMessage message;
   if (subscriber_->read(message, 1)) {
     const cv::Mat camera_frame = get_image_mat(message);
-    const std::vector<MarkerDetection> marker_detections =
-        detect_markers(camera_frame);
-
-    for (const auto & detection : marker_detections) {
-      std::cout << detection.marker_center_from_camera.translation().norm() << std::endl;
+    std::optional<SE3> board_from_camera = detect_board(camera_frame);
+    if(board_from_camera){
+      // publish a message using *board_from_camera
```

Comment:
This comment is a lie

Reply:
fiducial_from_camera_log = {board_from_camera.log()}; (maybe?)


But unpacking it is much easier:


`board_from_camera = SE3::exp(Eigen::Map<jcc::Vec6>>(array));`